cmd/machine-config-daemon/start.go

```diff
@@ -58,35 +51,6 @@ func init() {
 	startCmd.PersistentFlags().StringVar(&startOpts.promMetricsURL, "metrics-url", "127.0.0.1:8797", "URL for prometheus metrics listener")
 }
 
-func selfCopyToHost() error {
-	selfExecutableFd, err := os.Open("/proc/self/exe")
-	if err != nil {
-		return fmt.Errorf("opening our binary: %w", err)
-	}
-	defer selfExecutableFd.Close()
-	if err := os.MkdirAll(filepath.Dir(daemonconsts.HostSelfBinary), 0o755); err != nil {
-		return err
-	}
-	t, err := renameio.TempFile(filepath.Dir(daemonconsts.HostSelfBinary), daemonconsts.HostSelfBinary)
-	if err != nil {
-		return err
-	}
-	defer t.Cleanup()
-	var mode os.FileMode = 0o755
-	if err := t.Chmod(mode); err != nil {
-		return err
-	}
-	_, err = io.Copy(bufio.NewWriter(t), selfExecutableFd)
-	if err != nil {
-		return err
-	}
-	if err := t.CloseAtomicallyReplace(); err != nil {
-		return err
-	}
-	klog.Infof("Copied self to /run/bin/machine-config-daemon on host")
-	return nil
-}
-
 func runStartCmd(_ *cobra.Command, _ []string) {
```

Comment:
might be too much of a functionality change but could this be a `runE` rather than a `run` with cobra so we can return an error rather than klog.Fatal ?